src/cc/BPF.h

```diff
@@ -0,0 +1,162 @@
+/*
+ * Copyright (c) 2016 Facebook, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <cctype>
+#include <memory>
+#include <sstream>
+#include <string>
+
+#include "bcc_syms.h"
+#include "bpf_module.h"
+#include "BPFTable.h"
+#include "compat/linux/bpf.h"
+#include "libbpf.h"
+
+namespace ebpf {
+
+enum class bpf_attach_type {
+  probe_entry,
+  probe_return
+};
+
+struct open_probe_t {
+  void* reader_ptr;
+  std::string func;
+};
+
+class BPF {
```

Comment:
I'd recommend running clang-format on everything in this commit for stylistic consistency